CiJenkinsfile

```diff
@@ -0,0 +1,193 @@
+import java.text.SimpleDateFormat
+
+
+pipeline {
+    agent {
+        kubernetes {
+            label "builder-pod-${UUID.randomUUID().toString()}"
+            defaultContainer 'jnlp'
+            yaml """
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    some-label: builder-pod
+spec:
+  containers:
+  - name: helm
+    image: vfarcic/helm:2.8.2
+    command:
+    - cat
+    tty: true
+  - name: golang
+    image: golang:1.9
+    command: ["cat"]
+    tty: true
+  - name: git
+    image: alpine/git
+    command:
+    - cat
+    tty: true
+  - name: git
+    image: digitalinside/gren:latest
+    command:
+    - cat
+    tty: true    
+  - name: kubectl
+    image: vfarcic/kubectl
+    command: ["cat"]
+    tty: true
+  - name: docker
+    image: docker
+    command:
+    - cat
+    tty: true
+    volumeMounts:
+    - mountPath: /var/run/docker.sock
+      name: docker-sock-volume
+  volumes:
+  - name: docker-sock-volume
+    hostPath:
+      path: /var/run/docker.sock
+      type: File
+"""
+        }
+    }
+
+    parameters {
+        string(name: 'project', defaultValue: 'go-demo-3')
+        string(name: 'image', defaultValue: 'digitalinside/go-demo-3')
+        string(name: 'domain', defaultValue: '192.168.0.10.nip.io')
+        string(name: 'addr', defaultValue: 'go-demo-3.192.168.0.10.nip.io')
+        string(name: 'cmAddr', defaultValue: 'cm.4.192.168.0.10.nip.io')
+        string(name: 'chartVer', defaultValue: '0.0.1')
+    }
+
+    options {
+        timeout(time: 45, unit: 'MINUTES')
+    }
+
+    stages {
+        stage("prep") {
+            steps {
+                script {
+                    //https://issues.jenkins-ci.org/browse/JENKINS-52623 i found issue and waiting it to be fixed, i think carlos is on it now
+                    container('git') {
+                        env.GIT_COMMIT = sh (script: """ git rev-parse HEAD """, returnStdout: true).trim()
+                        env.shortGitCommit = "${env.GIT_COMMIT[0..10]}"
+                        env.BUILD_TAG = ciBuildVersionRead()
+                        echo "build tag set to: ${env.BUILD_TAG}"
+                    }
+
+                    currentBuild.displayName = new SimpleDateFormat("yy.MM.dd").format(new Date()) + "-" + env.BUILD_NUMBER
+                }
+            }
+        }
+
+
+        stage('build') {
+            steps {
+                container('docker') {
+                    //for feature branch
+                    ciK8sBuildImage(params.image, false, env.BUILD_TAG)
```

Comment:
I'd keep names of libraries agnostic to whether it's CDP, CD, or CI. If it's the same as used in other chapters, keep it as-is. If it's different, maybe rename it to something that describes what it does, not in which type of pipeline it'll be used.